tests/scripts/thread-cert/Cert_7_1_06_BorderRouterAsLeader.py

```diff
@@ -0,0 +1,468 @@
+        self.nodes[ROUTER_1].register_netdata()
+        self.simulator.go(10)
+
+        dut_addr = self.nodes[LEADER].get_addr(PREFIX_2001)
+        self.assertTrue(self.nodes[ROUTER_1].ping(dut_addr))
+        self.simulator.go(1)
+        self.assertTrue(self.nodes[SED].ping(dut_addr))
+
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        LEADER_RLOC = pv.vars['LEADER_RLOC']
+        LEADER_RLOC16 = pv.vars['LEADER_RLOC16']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ROUTER_1_RLOC16 = pv.vars['ROUTER_1_RLOC16']
+        ROUTER_1_RLOC = pv.vars['ROUTER_1_RLOC']
+        ROUTER_2 = pv.vars['ROUTER_2']
+        ROUTER_2_RLOC16 = pv.vars['ROUTER_2_RLOC16']
+        SED = pv.vars['SED']
+        MED = pv.vars['MED']
+        GUA = {}
+
+        for node in ('LEADER', 'ROUTER_1', 'SED'):
+            for addr in pv.vars['%s_IPADDRS' % node]:
+                if addr.startswith(Bytes(PREFIX_2001[:-5])):
+                    GUA[node] = addr
+
+        # Step 1: Ensure topology is formed correctly
+        pv.verify_attached('ROUTER_1', 'LEADER')
+        pv.verify_attached('ROUTER_2', 'LEADER')
+        pv.verify_attached('MED', 'LEADER', 'MTD')
+        pv.verify_attached('SED', 'LEADER', 'MTD')
+
+        # Step 2,3: Router_1 and Router_2  MUST send a CoAP Server Data
+        #           Notification frame to the Leader including the server’s
+        #           information(Prefix, Border Router):
+        #             CoAP Request URI
+        #                 coap://[<Leader address>]:MM/a/sd
+        #             CoAP Payload
+        #                 Thread Network Data TLV
+
+        # Step 4: Leader sends a CoAP ACK frame to each of Router_1 and
+        #         Router_2
+        for i in (1, 2):
+            with pkts.save_index():
+                pkts.filter_wpan_src64(pv.vars['ROUTER_%d' %i]).\
+                    filter_wpan_dst16(LEADER_RLOC16).\
+                    filter_coap_request(SVR_DATA_URI).\
+                    filter(lambda p:
+                           [Ipv6Addr(PREFIX_2001[:-3])] ==
+                           p.thread_nwd.tlv.prefix and\
+                           [pv.vars['ROUTER_%d_RLOC16' %i]] ==
+                           p.thread_nwd.tlv.border_router_16
+                           ).\
+                    must_next()
+                pkts.filter_wpan_src64(LEADER).\
+                    filter_ipv6_dst(pv.vars['ROUTER_%d_RLOC' %i]).\
+                    filter_coap_ack(SVR_DATA_URI).\
+                    must_next()
+
+        # Step 5: Leader MUST multicast MLE Data Response with the new
+        #         information collected from Router_1 and Router_2,
+        #         including the following TLVs:,
+        #              - Source Address TLV
+        #              - Leader Data TLV
+        #                  - Data Version field <incremented>
+        #                  - Stable Data Version field <incremented>
```

Comment:
how do we check for increasing Data Version?

Reply:
added data version check with previous packet before network data changes